exporters/prometheus/exporter.go

```diff
@@ -298,28 +299,38 @@ func addGaugeMetric[N int64 | float64](ch chan<- prometheus.Metric, gauge metric
 }
 
 // getAttrs parses the attribute.Set to two lists of matching Prometheus-style
-// keys and values. It sanitizes invalid characters and handles duplicate keys
-// (due to sanitization) by sorting and concatenating the values following the spec.
+// keys and values.
 func getAttrs(attrs attribute.Set, ks, vs [2]string, resourceKV keyVals) ([]string, []string) {
-	keysMap := make(map[string][]string)
-	itr := attrs.Iter()
-	for itr.Next() {
-		kv := itr.Attribute()
-		key := strings.Map(sanitizeRune, string(kv.Key))
-		if _, ok := keysMap[key]; !ok {
-			keysMap[key] = []string{kv.Value.Emit()}
-		} else {
-			// if the sanitized key is a duplicate, append to the list of keys
-			keysMap[key] = append(keysMap[key], kv.Value.Emit())
-		}
-	}
-
 	keys := make([]string, 0, attrs.Len())
 	values := make([]string, 0, attrs.Len())
-	for key, vals := range keysMap {
-		keys = append(keys, key)
-		slices.Sort(vals)
-		values = append(values, strings.Join(vals, ";"))
+	itr := attrs.Iter()
+
+	if model.NameValidationScheme == model.UTF8Validation {
+		// Do not perform sanitization if prometheus supports UTF-8.
+		for itr.Next() {
+			kv := itr.Attribute()
+			keys = append(keys, string(kv.Key))
+			values = append(values, kv.Value.Emit())
+		}
+	} else {
+		// It sanitizes invalid characters and handles duplicate keys
+		// (due to sanitization) by sorting and concatenating the values following the spec.
+		keysMap := make(map[string][]string)
+		for itr.Next() {
+			kv := itr.Attribute()
+			key := strings.Map(sanitizeRune, string(kv.Key))
```

Comment:
This call will always convert to underscores rather than follow the specified escaping scheme.  common.model has a function that will do this for you: https://github.com/prometheus/common/blob/d98411bbf37c7f936cef6d32a6ca965588521b4d/model/metric.go#L277-L339.

I'm not familiar with this code so I'm not sure if we can drop it in here.  The intent is that we should only have to write the escaping logic in common rather than reimplement it elsewhere.

Reply:
This change is currently just moving the existing escaping logic to the non-UTF-8 case.  I initially tried removing this escaping logic entirely, but found that I was getting validation errors.

Re: switching to the common library, this currently handles label collisions when converting to underscores by appending values together, separated by `;`.  Do you think we should drop that additional handling?